src/lib/api/util/format.ts

```diff
@@ -0,0 +1,64 @@
+import _Big from "big.js"
+import _Decimal from "decimal.js-light"
+import invariant from "tiny-invariant"
+import toFormat from "toformat"
+
+const Decimal = toFormat(_Decimal)
+const Big = toFormat(_Big)
+
+export enum Rounding {
+  ROUND_DOWN,
+  ROUND_HALF_UP,
+  ROUND_UP,
+}
+
+const toSignificantRounding = {
+  [Rounding.ROUND_DOWN]: Decimal.ROUND_DOWN,
+  [Rounding.ROUND_HALF_UP]: Decimal.ROUND_HALF_UP,
+  [Rounding.ROUND_UP]: Decimal.ROUND_UP,
+}
+
+const toFixedRounding = {
+  [Rounding.ROUND_DOWN]: Big.roundDown,
+  [Rounding.ROUND_HALF_UP]: Big.roundHalfUp,
+  [Rounding.ROUND_UP]: Big.roundUp,
+}
+
+export const toSignificant = (
+  value: string,
+  significantDigits: number,
+  format: object = { groupSeparator: "" },
+  rounding: Rounding = Rounding.ROUND_HALF_UP
+): string => {
+  invariant(!isNaN(Number(value)), `${value} is not an number.`)
+  invariant(
+    Number.isInteger(significantDigits),
+    `${significantDigits} is not an integer.`
+  )
+  invariant(significantDigits > 0, `${significantDigits} is not positive.`)
+
+  Decimal.set({
+    precision: significantDigits + 1,
+    rounding: toSignificantRounding[rounding],
+  })
+  const quotient = new Decimal(value).toSignificantDigits(significantDigits)
+  return quotient.toFormat(quotient.decimalPlaces(), format)
+}
+
+export const toFixed = (
+  value: string,
+  decimalPlaces: number,
+  format: object = { groupSeparator: "" },
+  rounding: Rounding = Rounding.ROUND_HALF_UP
+): string => {
+  invariant(!isNaN(Number(value)), `${value} is not an number.`)
+  invariant(
+    Number.isInteger(decimalPlaces),
+    `${decimalPlaces} is not an integer.`
+  )
+  invariant(decimalPlaces >= 0, `${decimalPlaces} is negative.`)
+
+  Big.DP = decimalPlaces
```

Comment:
I think you can just do something simple like this to round to 4 decimal places:

```ts
// value = 824213281784279560
const formattedValue = ethers.utils.formatEther(value);
// formattedValue = 0.82421328178427956
return Math.round(res * 1e4) / 1e4;
// return 0.8242
```

Reply:
We can certainly do it Math.round for this particular case but the method I'm trying to add here is something generic that we can use for any number format we could meet inside the project. Bring more flexibility, specify the group or the decimal separators, change the decimal number etc in that method.